### ISSUE TITLE
`find_fundamental`: a correspondence with weight 0 still changes F through the Hartley normalisation

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#4875]
`find_fundamental(points1, points2, weights)` (8POINT) documents per-correspondence weights and applies them to the DLT normal matrix (`run_8point`, `fundamental.py:305-313`; negative weights are clamped to 0). The Hartley normalisation that precedes the DLT (`fundamental.py:284-285`) is computed from all points without the weights, so a correspondence with weight 0 still moves the centroid and the scale of the normalised coordinates, and with them the solution. In the repro, 12 correspondences with 1-px noise give a mean Sampson error of 0.75 px² on themselves; adding one far-away pair with weight 0 changes `F` by 97% (relative, float64) and raises the mean Sampson error on the same 12 to 4.09 px². A weight of 0 is expected to be equivalent to dropping the correspondence, which is what a caller who pads point sets of different sizes into one batch relies on; Hartley–Zisserman's normalisation (§4.4.4) is defined on the points that enter the estimate.

## Repro

```python
import torch
from kornia.geometry.conversions import axis_angle_to_rotation_matrix
from kornia.geometry.epipolar import find_fundamental, sampson_epipolar_distance

K1 = torch.tensor([[800.0, 0, 320], [0, 760, 200], [0, 0, 1]], dtype=torch.float64)
K2 = torch.tensor([[700.0, 0, 300], [0, 740, 240], [0, 0, 1]], dtype=torch.float64)
R = axis_angle_to_rotation_matrix(torch.tensor([[0.1, -0.2, 0.05]], dtype=torch.float64))[0]
t = torch.tensor([[0.5], [0.1], [0.02]], dtype=torch.float64)
torch.manual_seed(0)
X = torch.rand(12, 3, dtype=torch.float64) * 2 - 1
X[:, 2] += 5
x1 = (K1 @ X.T).T
x2 = (K2 @ (R @ X.T + t)).T
x1 = (x1[:, :2] / x1[:, 2:])[None]
x2 = (x2[:, :2] / x2[:, 2:])[None] + torch.randn(1, 12, 2, dtype=torch.float64)  # 1 px noise in image 2
# the same 12 correspondences plus one far-away pair that is given weight 0
x1o = torch.cat([x1, torch.tensor([[[5000.0, -3000.0]]], dtype=torch.float64)], 1)
x2o = torch.cat([x2, torch.tensor([[[-2500.0, 4000.0]]], dtype=torch.float64)], 1)
w = torch.ones(1, 13, dtype=torch.float64)
w[0, 12] = 0.0

for dt in (torch.float64, torch.float32):
    a, b, ao, bo, wo = (v.to(dt) for v in (x1, x2, x1o, x2o, w))
    F12 = find_fundamental(a, b)                 # the 12 correspondences alone
    F13 = find_fundamental(ao, bo, weights=wo)   # the 12 + one pair with weight 0
    print(dt, "rel ||F13 - F12||:", f"{((F13 - F12).norm() / F12.norm()).item():.3g}",
          "| mean Sampson on the 12: F12", f"{sampson_epipolar_distance(a, b, F12).mean().item():.4g}",
          "F13", f"{sampson_epipolar_distance(a, b, F13).mean().item():.4g}")
# control: weights=None and all-ones weights agree bitwise, so the weight does reach the DLT
print("weights=None vs all-ones, max|diff|:", (find_fundamental(x1, x2) - find_fundamental(x1, x2, torch.ones(1, 12, dtype=torch.float64))).abs().max().item())
```

Output:

```text
torch.float64 rel ||F13 - F12||: 0.966 | mean Sampson on the 12: F12 0.7512 F13 4.089
torch.float32 rel ||F13 - F12||: 0.538 | mean Sampson on the 12: F12 0.7516 F13 5.244
weights=None vs all-ones, max|diff|: 0.0
```

kornia `main` 7ddf731bd, torch 2.14.0 (and 2.5.1: float64 line and control the same; float32 line `rel ||F13 - F12||: 0.37 | mean Sampson on the 12: F12 0.7517 F13 6.167`), CPU float64 and float32, macOS arm64.

## Classification

Silent wrong result. No idiom of 1–9; candidate: a weighted estimate normalised with unweighted statistics.

## Disposition

Focused fix, welcome as a PR: compute the centroid and the mean radius of the normalisation with the clamped weights (for example a private weighted variant of `normalize_points`), so that weight 0 and dropping the pair give the same `F`. `weights=None` and uniform weights give the same statistics as today; `RANSAC` passes all-ones weights (`ransac.py:215`, `:308`) and is unaffected.

Found by the batch-8 conventions audit (two-view geometry); a `test_wart_*` pin in the docs PR flips when the fix lands.

Posted on behalf of @ducha-aiki by Claude (Fable 5.1).
